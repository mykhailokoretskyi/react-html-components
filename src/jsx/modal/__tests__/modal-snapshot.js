jest.mock('react/lib/ReactDefaultInjection');
import React from 'react';
import Modal from '../modal';
import ModalContent from '../modalContent';
import ModalFooter from '../modalFooter';
import renderer from 'react/lib/ReactTestRenderer';

const defaultModal = (
    <Modal>
        <ModalContent>
            This is a default modal
        </ModalContent>
        <ModalFooter>
            With footer
        </ModalFooter>
    </Modal>
);

const fixedFooterModal = (
    <Modal type="modal-fixed-footer">
        <ModalContent>
            This is a modal
        </ModalContent>
        <ModalFooter>
            With fixed footer
        </ModalFooter>
    </Modal>
);

const bottomSheetModal = (
    <Modal type="bottom-sheet">
        <ModalContent>
            This is a bottom sheet modal
        </ModalContent>
        <ModalFooter>
            With footer
        </ModalFooter>
    </Modal>
);

describe('Modal', () => {
    it('default matches snapshot', () => {
        const component = renderer.create(defaultModal);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('fixed footer matches snapshot', () => {
        const component = renderer.create(fixedFooterModal);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('bottom sheet matches snapshot', () => {
        const component = renderer.create(bottomSheetModal);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
