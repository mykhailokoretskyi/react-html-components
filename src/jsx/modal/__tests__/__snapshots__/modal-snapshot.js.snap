exports[`Modal bottom sheet matches snapshot 1`] = `
<div
  className="modal bottom-sheet">
  <div
    className="modal-content">
    This is a bottom sheet modal
  </div>
  <div
    className="modal-footer">
    With footer
  </div>
</div>
`;

exports[`Modal default matches snapshot 1`] = `
<div
  className="modal ">
  <div
    className="modal-content">
    This is a default modal
  </div>
  <div
    className="modal-footer">
    With footer
  </div>
</div>
`;

exports[`Modal fixed footer matches snapshot 1`] = `
<div
  className="modal modal-fixed-footer">
  <div
    className="modal-content">
    This is a modal
  </div>
  <div
    className="modal-footer">
    With fixed footer
  </div>
</div>
`;
